src/index.ts

```diff
@@ -1,113 +1,147 @@
 import * as qs from 'qs';
+
 import { DOMAIN, MY_ACCOUNT, VAULT } from './endpoints';
 
-interface Config {
+interface IConfig {
   clientId: string;
-  isTestEnvironment?: boolean;
   scope: string[];
+  isTestEnvironment?: boolean;
   redirectUri?: string;
-  continueTo?: string;
-  responseType?: string;
+  continue?: string;
+  continueTo?: string; // NOTE: deprecated, remove in the next minor release, 1.3.0
+  responseType?: 'code' | 'token';
   locale?: string;
   state?: string;
 }
 
-interface Params {
+export interface IParams {
   client_id: string;
   redirect_uri: string;
-  continue: string;
   response_type: string;
   scope: string;
-  back?: string;
+  continue?: string;
   locale?: string;
   state?: string;
 }
 
-interface VaultOptions {
+export interface IDomains {
+  vault: string;
+  myaccount: string;
+}
+
+interface IVaultOptions {
   backTo?: string;
   newTab?: boolean;
 }
 
-interface MyaccountOptions {
+interface IMyAccountOptions {
   backTo?: string;
   newTab?: boolean;
   email?: string;
   authPage?: string;
+  showAuthToggle?: boolean;
 }
 
-const encodeConfig = (configs: object): string => {
-  // My Account separate key/values using semicolon
-  return encodeURIComponent(qs.stringify(configs, { delimiter: ';' }));
+function removeOAuth2Params(params: IParams) {
+  const validParams = { ...params };
+  delete validParams.scope;
+  delete validParams.redirect_uri;
+  delete validParams.response_type;
+  delete validParams.state;
+
+  return validParams;
+}
+
+function encodeConfigWithParams(params: IParams, configs: { [k: string]: string | boolean | undefined }) {
+  const endcodedConfigs = qs.stringify(configs, { delimiter: ';', encode: false });
+  return qs.stringify({ ...params, configs: endcodedConfigs });
 }
 
 class LinkSDK {
-  private domains: { [name: string]: string }
-  private params: Params
+  private domains: IDomains;
+  private params: IParams;
 
-  init(config: Config): void {
-    if (!config.clientId) {
+  public init(configs: IConfig): void {
+    if (!configs.clientId) {
       throw new Error('Need a clientId to initialise');
     }
 
+    // NOTE: probably "continue" is a reserve word, typescript seems to prevent me to add default to continue value here
```

Comment:
That's why it was named `continueTo`. We shouldn't use `continue`, it is a reserved word.

Reply:
`continueTo` is confuses peoples, they are think it is a redirect uri or something like that. I got a bunch of people including external dev keep asking the same questions every times even if there is a doc

Reply:
Then we need to change the name